.github/workflows/ci.yaml

```diff
@@ -7,7 +7,7 @@ on: # rebuild any PRs and main branch changes
       - main
 
 concurrency:
-  group: ${{ github.head_ref }}
+  group: ${{ github.head_ref || github.run_id }}
```

Comment:
No. I think it makes sense to allow multiple concurrent runs on main. They will be testing different things, after all.

The point is to prevent spamming a million runs as someone updates their own branch.